pkg/agent/controller/networkpolicy/fqdn_test.go

```diff
@@ -584,3 +584,85 @@ func TestSyncDirtyRules(t *testing.T) {
 		})
 	}
 }
+
+func TestOnDNSResponse(t *testing.T) {
+	currentTime := time.Now()
+	tests := []struct {
+		name             string
+		existingDNSCache map[string]dnsMeta
+		responseIPs      map[string]ipWithTTL
+		lowestTTL        uint32
+		expectedIPs      map[string]time.Time
+	}{
+		{
+			name: "new IP added and old IP retained",
+			existingDNSCache: map[string]dnsMeta{
+				"fqdn-test-pod.lfx.test": {
+					responseIPs: map[string]ipWithTTL{
+						// sample IP with some TTL
+						"192.0.2.1": {ip: net.ParseIP("192.0.2.1"), expirationTime: currentTime.Add(10 * time.Second)},
+						// sample IP with time simulating expired time, i thought that using negative time will
+						// simulate expired time as it will always equate  to before when compared to any time.
+						"192.0.2.2": {ip: net.ParseIP("192.0.2.2"), expirationTime: currentTime.Add(-1 * time.Second)},
+					},
+				},
+			},
+			responseIPs: map[string]ipWithTTL{
+				// we get new IP
+				"192.0.2.3": {ip: net.ParseIP("192.0.2.3"), expirationTime: currentTime.Add(10 * time.Second)},
+				// and an exisiting IP
+				"192.0.2.1": {ip: net.ParseIP("192.0.2.1"), expirationTime: currentTime.Add(10 * time.Second)},
+			},
+			lowestTTL: 30,
+			expectedIPs: map[string]time.Time{
+				// old unexpired IP should continue to have its actual TTL.
+				"192.0.2.1": currentTime.Add(10 * time.Second),
+				// new ip should have new expirationTime that was passed (minTTL)
+				"192.0.2.3": currentTime.Add(10 * time.Second),
+			},
+		},
+		{
+			name: "old IP expired",
```

Comment:
this is a strict subset of the previous test, and doesn't need to be included

there are 2 possible approaches here:
1. have a single unit test case with multiple different IPs, ensuring that all the branches / cases of the `onDNSResponse` are tested
2. have multiple test cases, each testing a different branch / scenario

Usually, we prefer the second approach because it makes test failures easier to identify / troubleshoot.